bindings/matrix-sdk-crypto-js/src/verification.rs

```diff
@@ -243,30 +243,28 @@ impl Sas {
     /// This confirms that the short auth strings match on both sides.
     ///
     /// Does nothing if we’re not in a state where we can confirm the
-    /// short auth string, otherwise returns a `MacEventContent` that
-    /// needs to be sent to the server.
+    /// short auth string.
+    ///
+    /// Returns a `Promise` for an array of `OutgoingRequest`s.
     pub fn confirm(&self) -> Promise {
         let me = self.inner.clone();
 
         future_to_promise(async move {
             let (outgoing_verification_requests, signature_upload_request) = me.confirm().await?;
-            let outgoing_verification_requests = outgoing_verification_requests
+            let outgoing_verification_requests: Array = outgoing_verification_requests
```

Comment:
This is already an `Array`, as specified by `.collect::<Result<Array, _>>()?` hereinbelow. No need to repeat it here :-).

Reply:
It seemed like my IDE wasn't intelligent enough to figure it out, for some reason. I agree with you though, it should be unnecessary.

my feeling is: the extra type annotation does no harm here. I can remove it if you'd prefer, though.

Reply:
I'd personally remove it, since it's already there in the `collect()` signature, and it's more idiomatic to indicate the type only once. But maybe it's sufficient to help type inference the other way by removing the `::<Result<Array, _>>` that's below? (I don't think so, because it might be necessary to convert the `Array<Result<T, E>>` into a `Result<Array<T>, E>`, but worth trying!)

Reply:
> But maybe it's sufficient to help type inference the other way by removing the ::<Result<Array, _>> that's below?

Yeah, that didn't work! Nice idea though.

I'll remove the extra annotation.